### PR TITLE
[OAP]Add a script to build OAP 

### PR DESCRIPTION
**Pull request by yao531441**
Add a script to build OAP and update pom.xml to enable build remote shuffle,

### CODE REVIEW

File: oap-cache/oap/pom.xml [hunk 1]
```diff
@@ -22,18 +22,12 @@
     <relativePath>../../pom.xml</relativePath>
   </parent>
 
-  <artifactId>oap-cache</artifactId>
+  <artifactId>oap-cache-oap</artifactId>
```
Comment (winningsix) [inline oap-cache/oap/pom.xml:25]: Any better name than ```oap-cache-oap```? It's a legacy problem at first refactor.

File: dev/make-distribution.sh [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        echo "Start building with all"
+        prepare_memkind
+        prepare_cmake
+        prepare_vmemcache
+        cd $dev_path
+        cd ..
+        mvn clean -q -Ppersistent-memory -Pvmemcache -DskipTests package
+        ;;
+        *)
+        echo "You should choose a existing options"
```
Comment (winningsix) [inline dev/make-distribution.sh:176]: Typo: ```an existed```

**Comment by winningsix** [comment]
Could you also update the README? @yao531441

File: dev/make-distribution.sh (hunk 2 above)
Comment (yao531441) [reply dev/make-distribution.sh]: Fixed

File: oap-cache/oap/pom.xml (hunk 1 above)
Comment (yao531441) [reply oap-cache/oap/pom.xml:25]: Do you have any advice?I have no idea how to name it better.

**Comment by yao531441** [comment]
Add a README about the folder dev now.

File: oap-cache/oap/pom.xml (hunk 1 above)
Comment (winningsix) [reply oap-cache/oap/pom.xml:25]: OK, we can use it as the name for now.

**Comment by winningsix** [comment]
LGTM

**Pull request merged**